HiP-DataStore/Controllers/ReviewCommentsController.cs

```diff
@@ -0,0 +1,130 @@
+    {
+        private readonly EventStoreService _eventStore;
+        private readonly EntityIndex _entityIndex;
+        private readonly ReviewIndex _reviewIndex;
+        private readonly ReviewCommentIndex _reviewCommentIndex;
+        private readonly IMongoDbContext _db;
+
+        public ReviewCommentsController(EventStoreService eventStore, InMemoryCache cache, IMongoDbContext db)
+        {
+            _eventStore = eventStore;
+            _entityIndex = cache.Index<EntityIndex>();
+            _reviewIndex = cache.Index<ReviewIndex>();
+            _reviewCommentIndex = cache.Index<ReviewCommentIndex>();
+            _db = db;
+        }
+
+        [HttpGet("{id}")]
+        [ProducesResponseType(typeof(ReviewCommentResult), 200)]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(403)]
+        [ProducesResponseType(404)]
+        public IActionResult GetRreview(int id)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            if (!_reviewCommentIndex.Exists(id))
+                return NotFound(ErrorMessages.ReviewCommentNotFound(id));
+
+            var reviewComment = _db.Get<ReviewComment>((ResourceTypes.ReviewComment, id));
+
+            var result = new ReviewCommentResult(reviewComment);
+
+            return Ok(result);
+        }
+
+        /// <summary>
+        /// Adds a Comment to a review
+        /// </summary>
+        /// <param name="reviewId">The id of the review</param>
+        /// <param name="args"></param>
+        [HttpPost]
+        [ProducesResponseType(204)]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(403)]
+        [ProducesResponseType(404)]
+        public async Task<IActionResult> PostAsync(int reviewId, ReviewCommentArgs args)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            if (!_reviewIndex.Exists(reviewId))
+                return BadRequest(ErrorMessages.NoReviewWithIdExists(reviewId));
+
+            var reviewArgs = await _eventStore.EventStream.GetCurrentEntityAsync<ReviewArgs>(ResourceTypes.Review, reviewId);
+
+            var newReviewArgs = reviewArgs.CloneArgs();
+
+            var reviewCommentId = _reviewCommentIndex.NextId();
+
+            // Add comment id to review
+            if (newReviewArgs.Comments == null)
+                newReviewArgs.Comments = new List<int>();
+            newReviewArgs.Comments.Add(reviewCommentId);
+
+            var reviewComment = new ReviewComment(args)
+            {
+                ReviewId = reviewId
+            };
+
+            if (args.Approved)
+                newReviewArgs.Approved = ReviewHelper.IsReviewApproved(newReviewArgs.Comments, args.Approved, newReviewArgs.StudentsToApprove, User.Identity, _reviewCommentIndex);
+
+            await EntityManager.CreateEntityAsync(_eventStore, reviewComment, ResourceTypes.ReviewComment, reviewCommentId, User.Identity.GetUserIdentity());
```

Comment:
Something looks wrong to me here. In the ResourceTypes class you define that the ReviewCommentArgs should be the corresponding class for the ReviewComment resource type. But here you pass an ReviewComment object instead of a ReviewCommentArgs object